Jellyfin.Server/Program.cs

```diff
@@ -0,0 +1,307 @@
+                    using (Stream rscstr = typeof(Program).Assembly
+                        .GetManifestResourceStream("Jellyfin.Server.Resources.Configuration.logging.json"))
+                    using (Stream fstr = File.Open(configPath, FileMode.CreateNew))
+                    {
+                        await rscstr.CopyToAsync(fstr);
+                    }
+                }
+                var configuration = new ConfigurationBuilder()
+                    .SetBasePath(appPaths.ConfigurationDirectoryPath)
+                    .AddJsonFile("logging.json")
+                    .AddEnvironmentVariables("JELLYFIN_")
+                    .Build();
+
+                // Serilog.Log is used by SerilogLoggerFactory when no logger is specified
+                Serilog.Log.Logger = new LoggerConfiguration()
+                    .ReadFrom.Configuration(configuration)
+                    .Enrich.FromLogContext()
+                    .CreateLogger();
+            }
+            catch (Exception ex)
+            {
+                Serilog.Log.Logger = new LoggerConfiguration()
+                    .WriteTo.Console(outputTemplate: "[{Timestamp:HH:mm:ss}] [{Level:u3}] {Message:lj}{NewLine}{Exception}")
+                    .WriteTo.File(
+                        Path.Combine(appPaths.LogDirectoryPath, "log_.log"),
+                        rollingInterval: RollingInterval.Day,
+                        outputTemplate: "[{Timestamp:yyyy-MM-dd HH:mm:ss.fff zzz}] [{Level:u3}] {Message}{NewLine}{Exception}")
+                    .Enrich.FromLogContext()
+                    .CreateLogger();
+
+                Serilog.Log.Logger.Fatal(ex, "Failed to create/read logger configuration");
+            }
+        }
+
+        public static IImageEncoder GetImageEncoder(
+            ILogger logger, 
+            IFileSystem fileSystem, 
+            StartupOptions startupOptions, 
+            Func<IHttpClient> httpClient,
+            IApplicationPaths appPaths,
+            IEnvironmentInfo environment,
+            ILocalizationManager localizationManager)
+        {
+            if (!startupOptions.ContainsOption("-enablegdi"))
+            {
+                try
+                {
+                    return new SkiaEncoder(logger, appPaths, httpClient, fileSystem, localizationManager);
+                }
+                catch (Exception ex)
+                {
+                    logger.LogInformation("Skia not available. Will try next image processor. {0}", ex.Message);
+                }
+
+                try
+                {
+                    return new ImageMagickEncoder(logger, appPaths, httpClient, fileSystem, environment);
+                }
+                catch
+                {
+                    logger.LogInformation("ImageMagick not available. Will try next image processor.");
+                }
+            }
+
+            return new NullImageEncoder();

```

Comment:
There should be a log about null image encoder

Reply:
This hasn't changed, still the same logic as before.

Reply:
Why don't we make it slightly better then? ;)

Reply:
Added